maintain `contextual` status of planning results and trigger search based planning

mmandlis opened the pull request.

shans commented on runtime/ts/plan/plan-producer.ts:
```diff
@@ -27,14 +28,23 @@ export class PlanProducer {
     this.speculator = new Speculator();
+    this.searchStore = searchStore;
+    if (this.searchStore) {
+      this.searchStore = searchStore;
```
setting this.searchStore twice

mmandlis replied:
removed, thanks!

mmandlis commented:
Addressed comment & fix for a bug caught by the selenium test (yay!) 
the fix is a bit hacky with respect to the launcher arc plans, added a TODO to address once 0_6_0 is done.
